ceylon-mode.el

```diff
@@ -95,35 +95,42 @@
     st)
   "Syntax table for `ceylon-mode'.")
 
-(set-default 'tab-width 4)
+(set-default 'tab-width                     4)
+(set-default 'ceylon-return-point-on-indent nil)
 (defun ceylon-indent-line ()
   "Indent current line as Ceylon code."
-  (beginning-of-line)
-  (if (bobp) ; beginning of buffer?
-      (indent-line-to 0)
-    (let (cur-indent)
-      (save-excursion
-        (forward-line -1)
-        (while (and (looking-at "^[ \t]*$") (not (bobp))) ; skip over blank lines
-          (forward-line -1))
-        (setq cur-indent (current-indentation))
-        (let* ((start (line-beginning-position))
-               (end   (line-end-position))
-               (open-parens    (how-many "(" start end))
-               (close-parens   (how-many ")" start end))
-               (open-braces    (how-many "{" start end))
-               (close-braces   (how-many "}" start end))
-               (open-brackets  (how-many "\\[" start end))
-               (close-brackets (how-many "\\]" start end))
-               (balance (- (+ open-parens open-braces open-brackets)
-                           (+ close-parens close-braces close-brackets))))
-          (if (looking-at"[ \t]*\\(}\\|)\\|]\\)")
-              (setq balance (+ balance 1)))
-          (setq cur-indent (+ cur-indent (* balance tab-width)))))
-      (if (looking-at "[ \t]*\\(}\\|)\\|]\\)")
-          (setq cur-indent (- cur-indent tab-width)))
-      (if (>= cur-indent 0)
-          (indent-line-to cur-indent)))))
+  (let* ((cur-column (and ceylon-return-point-on-indent
+                          (current-column))))
+    (beginning-of-line)
+
+    (if (bobp) ; beginning of buffer?
+        (indent-line-to 0)
+      (let (cur-indent (old-indent (current-indentation)))
```

Comment:
This should be split across two lines so that it’s clear that two independent variables are being declared (I was pretty confused for a bit because `cur-indent` isn’t bound to a value).